### PR TITLE
Set up Problem Matcher for Steep

### PR DESCRIPTION
**Pull request by ybiquitous**
### Summary or Background

It is smarter than using a pipe (`|`). See below:
- https://github.com/actions/toolkit/blob/27f76dfe1afb2b7e5e679cd8e97192d34d8320e6/docs/commands.md#problem-matchers
- https://github.com/actions/toolkit/blob/27f76dfe1afb2b7e5e679cd8e97192d34d8320e6/docs/problem-matchers.md

### Issues or Pull Requests

None.

### Checklist

- [-] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

**Comment by ybiquitous** [comment]
👍🏼 

image

**Review by dev-yakuza — APPROVED** [review]
LGTM 👍  Thanks!

File: .github/workflows/steep.yml [hunk 1]
```diff
@@ -19,9 +19,8 @@ jobs:
       - uses: ruby/setup-ruby@v1
         with:
           bundler-cache: true
-      - run: |
-          bundle exec rake steep:check | ruby -pe 'sub(/^\+ (.+):(\d+):(\d+): (.+)$/, %q{::error file=\1,line=\2,col=\3::\4})'
-        shell: bash
+      - run: echo "::add-matcher::.github/steep-matcher.json"
```
Comment (radomirbosak) [inline .github/workflows/steep.yml:22]: How does echoing this line help?

Also, is the `steep-matcher.json` filename hardcoded in steep? I could not find any code referring to it.

Comment (ybiquitous) [reply .github/workflows/steep.yml:22]: `::add-matcher` and a matcher JSON is a specification of [`actions/toolkit`](https://github.com/actions/toolkit/blob/27f76dfe1afb2b7e5e679cd8e97192d34d8320e6/docs/commands.md#problem-matchers).
It is not related to Steep itself.

**Pull request merged**